packages/2d/src/components/Grid.ts

```diff
@@ -1,17 +1,91 @@
 import {Shape, ShapeProps} from './Shape';
-import {PossibleVector2, Vector2Signal} from '@motion-canvas/core/lib/types';
-import {initial, vector2Signal} from '../decorators';
-import {SignalValue} from '@motion-canvas/core/lib/signals';
+import {initial, signal, vector2Signal} from '../decorators';
+import {
+  SimpleSignal,
+  SignalValue,
+  PossibleVector2,
+  Vector2Signal,
+  map,
+} from '@motion-canvas/core';
 
 export interface GridProps extends ShapeProps {
+  /**
+   * {@inheritDoc Grid.spacing}
+   */
   spacing?: SignalValue<PossibleVector2>;
+  /**
+   * {@inheritDoc Grid.start}
+   */
+  start?: SignalValue<number>;
+  /**
+   * {@inheritDoc Grid.end}
+   */
+  end?: SignalValue<number>;
 }
 
+/**
+ * A node for drawing a two-dimensional grid.
+ *
+ * @preview
+ * ```tsx editor
+ * import {Grid, makeScene2D} from '@motion-canvas/2d';
+ * import {all, createRef} from '@motion-canvas/core';
+ *
+ * export default makeScene2D(function* (view) {
+ *   const grid = createRef<Grid>();
+ *
+ *   view.add(
+ *     <Grid
+ *       ref={grid}
+ *       width={'100%'}
+ *       height={'100%'}
+ *       stroke={'#666'}
+ *       start={0.5}
+ *       end={0.5}
+ *     />,
+ *   );
+ *
+ *   yield* all(
+ *     grid().end(1, 1).wait(1).to(0.5, 1),
+ *     grid().start(0, 1).wait(1).to(0.5, 1),
```

Comment:
When providing examples for Nodes, they should be initially visible and then hide/reappear
So that even if they don't press play, the node will be immediately visible.
```suggestion
 *       start={0}
 *       end={1}
 *     />,
 *   );
 *
 *   yield* all(
 *     grid().end(0.5, 1).to(1, 1).wait(1),
 *     grid().start(0.5, 1).to(0, 1).wait(1),
```